### PR TITLE
Support assembly as message origin

### PR DESCRIPTION
**Pull request by sbomer**
https://github.com/mono/linker/issues/2211 is caused by an assembly-level `MetadataUpdateHandlerAttribute` (which requires DAMT.All) on Microsoft.AspNetCore.Http.Abstractions. Currently this seems to be an analysis hole - we don't show warnings for assembly-level attributes because we don't consider the assembly (or assembly attribute) as a possible message origin. As far as I know we don't have a targeted suppression mechanism for such warnings, but we need to figure out how to close this hole.

I started working on an approach that will at least show such warnings.

Fixes https://github.com/mono/linker/issues/2211

### CODE REVIEW

**Comment by mateoatr** [comment]
> As far as I know we don't have a targeted suppression mechanism for such warnings

You mean having an assembly as the target of an UnconditionalSuppressMessage? If this is the case, I think we have avoided this since we don't want users to be able to cross-assembly suppress trimmer warnings.

**Comment by sbomer** [comment]
I agree we don't want cross-assembly suppressions - but what might be nice is some way to suppress warnings that come from the current assembly's attributes (without also suppressing the same warning for all other attributes in the same assembly). @vitek-karas agreed we don't need that for now though.

**Comment by sbomer** [comment]
@vitek-karas @mateoatr PTAL

File: src/linker/Linker.Steps/MarkStep.cs [hunk 1]
```diff
@@ -1362,7 +1368,7 @@ protected void MarkAssembly (AssemblyDefinition assembly, DependencyInfo reason)
 				return;
 
 			// We don't have a good origin for "assembly" level, so just use null for now
```
Comment (mateoatr) [inline src/linker/Linker.Steps/MarkStep.cs:1370]:
```suggestion
```

**Review by mateoatr — APPROVED** [review]
LGTM!

**Review by vitek-karas — APPROVED** [review]
Looks good

**Pull request merged**